### PR TITLE
docs: Add TechDocs and documentation

### PR DESCRIPTION
**Pull request by LeKeve**
This PR sets up a working TechDocs configuration for local dev and our docker image build. It also includes documentation on how to build new documentation projects for TechDocs.

[API-5210](https://vajira.max.gov/browse/API-5210)

My work points to a `/documentation/techdocs` project on master (which doesn't exist yet because it's contained on this branch), so I created a separate testing branch (`local-docs-testing`) that you can point backstage to. You will not be able to point to `API-5210/techdocs` because Backstage has issues parsing GitHub branches with `/` in their name. You should add the following to your `app-config.local.yaml` file to view this documentation project on your local instance of backstage:

```yaml
catalog:
  rules:
    - allow: [Component, API, Group, User, Template, Location]
  locations:
    - type: url
      # override default and point to local-docs-testing branch
      target: https://github.com/department-of-veterans-affairs/backstage/blob/local-docs-testing/documentation/techdocs/techdocs.yaml
```

### CODE REVIEW

**Comment by LeKeve** [comment]
This work is based on my previous PR. Pointing to that branch until it gets reviewed, then I'll update the base branch.

File: documentation/techdocs/docs/creating-docs.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# required plugin for techdocs generator
+plugins:
+  - techdocs-core
+
+# optional but recommended, if not provided
+# mkdocs will automatically generate basic navigation
+nav:
+  - Introduction: index.md
+  - Creating Documentation: creating-docs.md
```
Comment (rianfowler) [inline documentation/techdocs/docs/creating-docs.md:34]: does this support a tree or just a flat list structure?

File: documentation/techdocs/docs/creating-docs.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/docs
+  - index.md
+  - another-section.md
+mkdocs.yml
+documentation-name.yaml
+```
+
+> :warning: `/docs`, `mkdocs.yml`, and `index.md` are strict naming requirements and are required by MkDocs to generate your documentation
+
+The `/docs` directory contains the markdown files that make up your documentation. An `index.md` file is required as the initial entry point to your documentation project.
```
Comment (rianfowler) [inline documentation/techdocs/docs/creating-docs.md:17]:
```suggestion
### /docs directory 

The `/docs` directory contains the markdown files that make up your documentation. An `index.md` file is required as the initial entry point to your documentation project.
```

File: documentation/techdocs/docs/creating-docs.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+`mkdocs.yml` specifies how the documentation should be built and includes a `nav` property which allows you to configure docs navigation. Example below:
```
Comment (rianfowler) [inline documentation/techdocs/docs/creating-docs.md:19]:
```suggestion
### mkdocs.yml

`mkdocs.yml` specifies how the documentation should be built and includes a `nav` property which allows you to configure docs navigation. Example below:
```

File: documentation/techdocs/docs/creating-docs.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  - techdocs-core
+
+# optional but recommended, if not provided
+# mkdocs will automatically generate basic navigation
+nav:
+  - Introduction: index.md
+  - Creating Documentation: creating-docs.md
+```
+
+Finally, you should create a yaml file that contains your project configuration. This file includes your documentation name, description, and a reference to where your docs are stored.
```
Comment (rianfowler) [inline documentation/techdocs/docs/creating-docs.md:37]:
```suggestion
### [documentation-name].yml

Finally, you should create a yaml file that contains your project configuration. This file includes your documentation name, description, and a reference to where your docs are stored.
```
What is "documentation name"? Is that something I need to provide? is it like the title of my documentation?

File: documentation/techdocs/docs/adding-to-backstage.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+## Registering via Catalog Import
+
+You can also use the Catalog Import plugin to register a TechDocs project. Simply navigate to http://localhost:3000/catalog-import and follow the instructions in the UI.
+
+## Adding a TechDocs reference to a Service Catalog Component
+
+Each component in the Backstage Service Catalog can be documented via TechDocs.
+
+<img width="632" alt="Screen Shot 2021-02-18 at 7 59 26 PM" src="https://user-images.githubusercontent.com/9746156/108455935-0993c380-7224-11eb-9fa2-d845956bec07.png">
```
Comment (rianfowler) [inline documentation/techdocs/docs/adding-to-backstage.md:25]: Screenshots are 👍

**Review by rianfowler — APPROVED** [review]
This looks good. I haven't tested it yet but I think it's fine to merge. 

Suggested some additional headings on one of the pages to make scanning easier.

File: documentation/techdocs/docs/creating-docs.md (hunk 1 above)
Comment (LeKeve) [reply documentation/techdocs/docs/creating-docs.md:34]: Supports a tree structure as well:

[Example from backstage](https://github.com/backstage/backstage/blob/master/mkdocs.yml)

File: documentation/techdocs/docs/creating-docs.md (hunk 4 above)
Comment (LeKeve) [reply documentation/techdocs/docs/creating-docs.md:39]: Documentation name is the title of the documentation. I'll update that, documentation title is clearer.

**Pull request merged**